android/src/main/java/com/amplitude/android/Amplitude.kt

```diff
@@ -117,6 +121,22 @@ open class Amplitude(
         })
     }
 
+    private suspend fun migrateApiKeyStorages() {
+        val configuration = this.configuration as Configuration
+
+        val storage = storage as? AndroidStorage
+        if (storage != null) {
+            val apiKeyStorage = AndroidStorage(configuration.context, configuration.apiKey, logger, storage.prefix)
+            StorageKeyMigration(apiKeyStorage, storage, logger).execute()
+        }
+
+        val identifyInterceptStorage = identifyInterceptStorage as? AndroidStorage
+        if (identifyInterceptStorage != null) {
+            val apiKeyStorage = AndroidStorage(configuration.context, configuration.apiKey, logger, identifyInterceptStorage.prefix)
+            StorageKeyMigration(apiKeyStorage, identifyInterceptStorage, logger).execute()
+        }
+    }
+
```

Comment:
nit: To keep the new logic in Ampliutde.kt to a minium I think it would be nice to extract this logic to a separate class e.g. `ApiKeyStorageMigration(this).execute()` like `RemnantDataMigration`.

Reply:
Extracted